app/src/main/java/com/example/tasky/agenda/domain/data/sync/SyncTaskScheduler.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.tasky.agenda.domain.data.sync
+
+import com.example.tasky.agenda.domain.model.Task
+import com.example.tasky.core.data.database.SyncOperation
+import kotlin.time.Duration
+
+interface SyncTaskScheduler {
+
+    suspend fun scheduleSync(type: SyncType)
+    suspend fun cancelAllSyncs()
+
+    sealed interface SyncType {
+        data class FetchTasks(val interval: Duration) : SyncType
+        data class DeleteTask(val taskId: String) : SyncType
+        class UpsertTask(val task: Task, val operation: SyncOperation) : SyncType
```

Comment:
You could also generalize `UpsertTask`, `DeleteTask`, `FetchTasks`   into `UpsertAgendaItem`, `DeleteAgendaItem` and  `FetchAgendaItems`  then handle the branching by item kind inside the worker. That way the sealed interface stays lean and you avoid growing it with a separate operation and  type for every agenda item.